contracts/BC_fusion/StakeHub.sol

```diff
@@ -0,0 +1,1203 @@
+        // should not happen
+        emit unexpectedPackage(channelId, msgBytes);
+    }
+
+    /*----------------- external functions -----------------*/
+    /**
+     * @param consensusAddress the consensus address of the validator
+     * @param voteAddress the vote address of the validator
+     * @param blsProof the bls proof of the vote address
+     * @param commission the commission of the validator
+     * @param description the description of the validator
+     */
+    function createValidator(
+        address consensusAddress,
+        bytes calldata voteAddress,
+        bytes calldata blsProof,
+        Commission calldata commission,
+        Description calldata description
+    ) external payable whenNotPaused notInBlackList {
+        // basic check
+        address operatorAddress = msg.sender;
+        if (_validatorSet.contains(operatorAddress)) revert ValidatorExisted();
+        if (consensusToOperator[consensusAddress] != address(0) || _legacyConsensusAddress[consensusAddress]) {
+            revert DuplicateConsensusAddress();
+        }
+        if (voteToOperator[voteAddress] != address(0) || _legacyVoteAddress[voteAddress]) {
+            revert DuplicateVoteAddress();
+        }
+        bytes32 monikerHash = keccak256(abi.encodePacked(description.moniker));
+        if (_monikerSet[monikerHash]) revert DuplicateMoniker();
+
+        uint256 delegation = msg.value;
+        if (delegation < minSelfDelegationBNB + LOCK_AMOUNT) revert SelfDelegationNotEnough();
+
+        if (consensusAddress == address(0)) revert InvalidConsensusAddress();
+        if (
+            commission.maxRate > 5_000 || commission.rate > commission.maxRate
+                || commission.maxChangeRate > commission.maxRate
+        ) revert InvalidCommission();
+        if (!_checkMoniker(description.moniker)) revert InvalidMoniker();
+        // proof-of-possession verify
+        if (!_checkVoteAddress(voteAddress, blsProof)) revert InvalidVoteAddress();
+
+        // deploy stake credit proxy contract
+        address creditContract = _deployStakeCredit(operatorAddress, description.moniker);
+
+        _validatorSet.add(operatorAddress);
+        _monikerSet[monikerHash] = true;
+        Validator storage valInfo = _validators[operatorAddress];
+        valInfo.consensusAddress = consensusAddress;
+        valInfo.operatorAddress = operatorAddress;
+        valInfo.creditContract = creditContract;
+        valInfo.createdTime = block.timestamp;
+        valInfo.voteAddress = voteAddress;
+        valInfo.description = description;
+        valInfo.commission = commission;
+        valInfo.updateTime = block.timestamp;
+        consensusToOperator[consensusAddress] = operatorAddress;
+        voteToOperator[voteAddress] = operatorAddress;
+
+        emit ValidatorCreated(consensusAddress, operatorAddress, creditContract, voteAddress);
+        emit Delegated(operatorAddress, operatorAddress, delegation, delegation);
```

Comment:
I think not `delegation` but `delegation-LOCK_AMOUNT`.

Reply:
Do we need emit event for the dead account?

Reply:
Done